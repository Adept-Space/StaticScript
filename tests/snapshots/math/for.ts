
{
    function calculateTotalFromRange(start: number, end: number): number {
        console_log("calculateTotalFromRange");

        let total: number = 0;

        for (let i = start; i < end; i++) {
            total = total + i;
        }

        return total;
    }

    function simpleBreak(): number {
        console_log("simpleBreak");

        let total: number = 0;

        for (let i = 0; ; i++) {
            total = total + i;

            if (i > 100) {
                break;
            }
        }

        return total;
    }

    function simpleContinue(): number {
        console_log("simpleContinue");

        let total: number = 0;

        for (let i = 0; i < 100; i++) {
            if (i > 50) {
                continue;
            }

            total = total + i;
        }

        return total;
    }

    console_log(calculateTotalFromRange(1, 100));
    console_log(calculateTotalFromRange(50, 100));
    console_log(simpleBreak());
    console_log(simpleContinue());
}
