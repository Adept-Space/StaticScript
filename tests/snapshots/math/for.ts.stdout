calculateTotalFromRange
4950.000000
calculateTotalFromRange
3725.000000
simpleBreak
5151.000000
simpleContinue
1275.000000
